### DISCUSSION TITLE
Training model (LSTM) with exogenous variables

### DISCUSSION THREAD

**Discussion by labenech** [Nixtla/neuralforecast#792] [Q&A]
Hello,
I have 2 questions about the use of exogenous variables. 

I have a monthly frequency time series whose values are available up to month m and I would like to predict the values for the next 3 months (m+1, m+2, m+3). 
All my exogenous variables are available up to month m+1.
- should i leave hist_exog_list empty and put the list of all variables in futr_exog_list?
- for months m+2 and m+3, some exogenous variables have values not yet available and are at NaN. How should these NaN values be handled?

help appreciated
Thanks

**Comment by wasf84** [comment]
Hello, my friend.

"_* should i leave hist_exog_list empty and put the list of all variables in futr_exog_list?_"

The variables in ‘hist_exog_list’ are those that, by definition, are only known during the training of your model, up to time ‘t’. If they are variables that can be inferred (perhaps by some other type of model) in the forecast horizon of your work, then I believe they should be included in the ‘futr_exog_list’.

"_*for months m+2 and m+3, some exogenous variables have values not yet available and are at NaN. How should these NaN values be handled?_"

If, in the m+2 and m+3 horizons, the variables cannot be acquired, it would be best to implement an auxiliary model that forecasts this data to help populate the ‘futr_exog_list’. I am doing exactly that in a project I am working on.

I’m not sure if I’ve adequately addressed your questions. Let me know if it was helpful in any way.

Take care.

**translated to english by Copilot**
